src/aux-server/aux-web/shared/scene/decorators/DimensionPositionDecorator.ts

```diff
@@ -281,24 +281,89 @@ export class DimensionPositionDecorator extends AuxBot3DDecoratorBase {
 
                 const objWorld = new Vector3();
                 this._rotationObj.getWorldPosition(objWorld);
-                const direction = new CasualVector3(
-                    objWorld.x,
-                    objWorld.y,
-                    objWorld.z
-                ).subtract(
-                    new CasualVector3(
-                        cameraWorld.x,
-                        cameraWorld.y,
-                        cameraWorld.z
-                    )
+
+                let direction: CasualVector3;
+                let upVector: CasualVector3;
+
+                const transformer = getBotTransformer(calc, this.bot3D.bot);
+                const currentGridPos = getBotPosition(
+                    calc,
+                    this.bot3D.bot,
+                    this.bot3D.dimension
                 );
-                const lookRotation = new Rotation({
-                    direction: direction,
-                    upwards: new CasualVector3(
+                const coordinateTransform =
+                    this.bot3D.coordinateTransformer && !hasValue(transformer)
+                        ? this.bot3D.coordinateTransformer(currentGridPos)
+                        : null;
+
+                if (this._orientationMode === 'billboard') {
+                    direction = new CasualVector3(
+                        objWorld.x - cameraWorld.x,
+                        objWorld.y - cameraWorld.y,
+                        objWorld.z - cameraWorld.z
+                    );
+
+                    upVector = new CasualVector3(
                         this._rotationObj.up.x,
                         this._rotationObj.up.y,
                         this._rotationObj.up.z
-                    ),
+                    );
+                } else if (coordinateTransform) {
+                    const globeCenter = new Vector3(0, 0, 0); //Assuming globe center is at origin
+
+                    const globalDirection = new Vector3().subVectors(
+                        objWorld,
+                        globeCenter
+                    );
+
+                    const invTransform = new Matrix4()
+                        .copy(coordinateTransform)
+                        .invert();
+                    const localDirection = globalDirection
+                        .clone()
+                        .applyMatrix4(invTransform);
+
+                    const adjustedGlobalDirection =
+                        localDirection.applyMatrix4(coordinateTransform);
```

Comment:
@TroyceGowdy All this is unnecessary. `globalDirection` is the same as `objWorld`. `adjustedGlobalDirection` is also the same as `objWorld`. `localDirection` is the same as `0,0,0`. `invTransform` isn't used for anything useful.